back-end-side.UnitTests/ReaderTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using back_end_side.Controllers;
+using back_end_side.DbFiles;
+using back_end_side.Models;
+using back_end_side.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Record = back_end_side.Models.Record;
+
+namespace back_end_side.UnitTests
+{
+    public class ReaderTests
+    {
+        
+        [Fact]
+        public async Task GetFileData_SEBfileWithTwoRecordsIsGiven_SavesTwoRecordsToDatabase()
+        {
+            FileModel fileModel = new FileModel
+            {
+                FileName = "seb.csv",
+                FormFile = null
+            };
+
+            var context = GetDatabaseContext();
+            var reader = new ReportReader();
+            var uploadService = new UploadService(context, reader);
+
+            var codeBaseUrl = new Uri(Assembly.GetExecutingAssembly().CodeBase);
+            var codeBasePath = Uri.UnescapeDataString(codeBaseUrl.AbsolutePath);
+            var dirPath = Path.GetDirectoryName(codeBasePath);
+
+            var path = Path.Combine(dirPath, "testFiles\\seb.csv");
+
+            using (var stream = File.OpenRead(path))
+            {
+                var file = new FormFile(stream, 0, stream.Length, null, Path.GetFileName(stream.Name));
+                fileModel.FormFile = file;
+                await uploadService.GetFileData(fileModel);
+            }
+
+            var pathSw = Path.Combine(dirPath, "testFiles\\swedbank.csv");
+
+            using (var stream = File.OpenRead(pathSw))
+            {
+                var file = new FormFile(stream, 0, stream.Length, null, Path.GetFileName(stream.Name));
+                fileModel.FormFile = file;
+                fileModel.FileName = "swedbank.csv";
+                await uploadService.GetFileData(fileModel);
+            }
+
+            var pathP = Path.Combine(dirPath, "testFiles\\paysera.csv");
+
+            using (var stream = File.OpenRead(pathP))
+            {
+                var file = new FormFile(stream, 0, stream.Length, null, Path.GetFileName(stream.Name));
+                fileModel.FormFile = file;
+                fileModel.FileName = "paysera.csv";
+                await uploadService.GetFileData(fileModel);
+            }
+
+            var expected = "Firmine parduotuve\\Tilzes g. 133\\Siauliai\\76349     LTULTU";
+            var actual = context.Expenses.Find(1).Seller;
```

Comment:
Maybe it would be smarter to compare record with record instead of comparing Seller of record?

Reply:
Good point, I will change it